### PR TITLE
LPS-202986 The KB article with Expired status can not be deleted

### PR DESCRIPTION
**Pull request by marcogalluzzi**
From: https://liferay.atlassian.net/browse/LPS-202986

### CODE REVIEW

File: modules/apps/knowledge-base/knowledge-base-service/src/main/java/com/liferay/knowledge/base/service/impl/KBArticleLocalServiceImpl.java [hunk 1]
```diff
@@ -2614,7 +2614,7 @@ private void _startWorkflowInstance(
 
 	private void _updateKBArticleAsset(
```
Comment (AliciaGarciaGarcia) [inline modules/apps/knowledge-base/knowledge-base-service/src/main/java/com/liferay/knowledge/base/service/impl/KBArticleLocalServiceImpl.java:2615]: inline this method on another commit

**Review by AliciaGarciaGarcia — CHANGES_REQUESTED** [review]

File: modules/apps/knowledge-base/knowledge-base-service/src/main/java/com/liferay/knowledge/base/service/impl/KBArticleLocalServiceImpl.java (hunk 1 above)
Comment (marcogalluzzi) [reply modules/apps/knowledge-base/knowledge-base-service/src/main/java/com/liferay/knowledge/base/service/impl/KBArticleLocalServiceImpl.java:2615]: It's used twice

Comment (AliciaGarciaGarcia) [reply modules/apps/knowledge-base/knowledge-base-service/src/main/java/com/liferay/knowledge/base/service/impl/KBArticleLocalServiceImpl.java:2615]: one directly from the `updateKBArticleAsset` and the other can call directly to the that.

no?

Comment (marcogalluzzi) [reply modules/apps/knowledge-base/knowledge-base-service/src/main/java/com/liferay/knowledge/base/service/impl/KBArticleLocalServiceImpl.java:2615]: You are right! I missed that.

**Comment by marcogalluzzi** [comment]
ci:test:sf

**Review by AliciaGarciaGarcia — APPROVED** [review]

**Comment by AliciaGarciaGarcia** [comment]
ci:forward

**Comment by SylviaLuan** [comment]
ci:test:stable

**Comment by SylviaLuan** [comment]
Hi @marcogalluzzi,

There is a failure that caused by this pull request.

Step to reproduce:

- Add a KB Article and Subscribe it
- Add a new user with a administrator role
- Log out and login with new user
- EXPIRE the KB article
- Logout and Login with test
- Go to Notifications 

Expect result: Can see A knowledge base article has expired
Actaul result: The message is Notification no longer applies. Notification for Knowledge Base was deleted.

**Comment by marcogalluzzi** [comment]
ci:test:sf

**Comment by marcogalluzzi** [comment]
Hi @AliciaGarciaGarcia please review last commit thanks

File: modules/apps/knowledge-base/knowledge-base-web/src/main/java/com/liferay/knowledge/base/web/internal/asset/model/KBArticleAssetRendererFactory.java [hunk 2]
```diff
@@ -151,6 +151,15 @@ private KBArticle _getKBArticle(long classPK, int status)
 			return kbArticle;
 		}
 
+		if (status == WorkflowConstants.STATUS_APPROVED) {
```
Comment (AliciaGarciaGarcia) [inline modules/apps/knowledge-base/knowledge-base-web/src/main/java/com/liferay/knowledge/base/web/internal/asset/model/KBArticleAssetRendererFactory.java:154]: I feel odd the "if is approved find the expired"

what do you think removing the if:
`	if (status == WorkflowConstants.STATUS_APPROVED) {`

**Review by AliciaGarciaGarcia — CHANGES_REQUESTED** [review]

File: modules/apps/knowledge-base/knowledge-base-web/src/main/java/com/liferay/knowledge/base/web/internal/asset/model/KBArticleAssetRendererFactory.java (hunk 2 above)
Comment (marcogalluzzi) [reply modules/apps/knowledge-base/knowledge-base-web/src/main/java/com/liferay/knowledge/base/web/internal/asset/model/KBArticleAssetRendererFactory.java]: You are right. It looked strange. I made a little refactoring and now it looks better, I hope. See my last commit.

**Comment by marcogalluzzi** [comment]
ci:test:sf

**Comment by SylviaLuan** [comment]
ci:test:relevant

**Comment by SylviaLuan** [comment]
ci:test:stable

File: modules/apps/knowledge-base/knowledge-base-web/src/main/java/com/liferay/knowledge/base/web/internal/asset/model/KBArticleAssetRendererFactory.java (hunk 2 above)
Comment (AliciaGarciaGarcia) [reply modules/apps/knowledge-base/knowledge-base-web/src/main/java/com/liferay/knowledge/base/web/internal/asset/model/KBArticleAssetRendererFactory.java]: 👍🏻

**Review by AliciaGarciaGarcia — APPROVED** [review]

**Comment by AliciaGarciaGarcia** [comment]
ci:forward

**Comment by AliciaGarciaGarcia** [comment]
ci:forward:force

**Pull request closed without merging**